SETUP/GRAPHS.md

```diff
@@ -0,0 +1,174 @@
+# Graphs
+
+This file contains a description of the way graphs are created in dproofreaders.
+
+In case of problems, please post to the
+[DP Site Code](http://www.pgdp.net/phpBB3/viewforum.php?f=32) forum at the
+[primary DP site](https://www.pgdp.net/c).
+
+## Overview
+Graphs are drawn on the browser using a set of functions from
+[svgGraphs.js](../scripts/svgGraphs.js) that interact with the
+[d3js](https://d3js.org/) library to render
+[Scalable Vector Graphics (svg)](https://en.wikipedia.org/wiki/Scalable_Vector_Graphics)
+elements. The 4 types of graphs that are supported are
+* Bar
+* Line
+* Pie
+* Stacked Area
+
+## Usage
+
+These graphs are configured using a plain old javascript object. To facilitate
+creating the javascript to render a graph from PHP, a PHP utility method
+[build_svg_graph_inits](../pinc/graph_data.inc) was created. It is a function
+that takes an array of arrays representing the graphs to render. Each graph to
+render is a PHP array where the entries are
+
+* type - type of graph to be rendered. Either `barLineGraph`, `pieGraph`, or
+`stackedAreaGraph`
+* id - id of the div on the page to render the graph into
+* configuration - PHP array configuring the graph. Expanded on below.
+
+Example Usage:
+
+```php
+build_svg_graph_inits([["barLineGraph", "myGraphId", $graphConfig]]);
+```
+
+This will give you a string you can use to pass to js_data in the
+[output_header](../pinc/theme.inc) function. Something like
+
+```php
+"$(function(){barLineGraph('myGraphId', graphConfigEncodedAsJSON)});"
+```
+
+### Embedded scenarios
+
+In some cases, you may wish to embed a graph as if it were an image. Examples
+include site news items or other cases where you may not be able to output
+scripts to draw the graphs. In these scenarios, we recommend you create a
+standalone page that renders just the graph, and then embed that page as an
+iframe. You may want to style the iFrame to remove any iFrame borders to have a
+seamless image like experience.
+
+One such page for example of a page designed for being embedded as an iFrame is
+[round_backlog_days.php](../stats/round_backlog_days.php). It can be embedded
+in a news item with the following html.
+
+```html
+<iframe src="/stats/round_backlog_days.php" style="width:320px;height:205px;border:0;">
```

Comment:
I'm going to be picky again, but you're welcome to ignore me, because I know that the code works either way.

How should attributes be styled in the code? I.e. spaced for easier reading, or all squeezed together. I didn't notice this in the code before seeing it smushed together here, but in looking at the code, I see that's how it would be written out. At the same time, some of the other attributes have a space both between the attribute and the value, and following the semicolon.

I didn't notice anything talking about that in the style guide, so I guess partly I'm asking if there's a particular format we should aim for.

That being said, everything looks good to me other than that inconsistency.

Reply:
I think it depends on what your goal is. Removing spaces makes the HTML smaller (barely) but reduces legibility. Personally I think the spaces should probably be there since we don't aggressively attempt to make our HTML, JS, nor CSS minimal.

Reply:
That being said, I can update this in a subsequent MR when I fix the min width of bars.

Reply:
I prefer legibility, since it's not a significant reduction in size.

Reply:
went ahead and implemented the change since this wasn't merged yet.